src/main/java/org/terasology/launcher/gui/javafx/SettingsController.java

```diff
@@ -282,13 +198,12 @@ protected void updateInitialHeapSizeBox() {
         }
     }
 
-    void initialize(final Path newLauncherDirectory, final Path newDownloadDirectory, final BaseLauncherSettings newLauncherSettings,
-                    final TerasologyGameVersions newGameVersions, final Stage newStage) {
-        this.launcherDirectory = newLauncherDirectory;
-        this.downloadDirectory = newDownloadDirectory;
-        this.launcherSettings = newLauncherSettings;
-        this.gameVersions = newGameVersions;
-        this.stage = newStage;
+    void update(final LauncherConfiguration config) {
+        this.launcherDirectory = config.getLauncherDirectory();
+        this.downloadDirectory = config.getDownloadDirectory();
+        this.launcherSettings = config.getLauncherSettings();
+        this.gameVersions = config.getGameVersions();
+        //root.setDisable(false);
```

Comment:
as mentioned above, please don't check in commented code. Either discard it or use a feature toggle.